### ISSUE TITLE
Stache #each with an object doesn't update when keys are replaced

### ISSUE THREAD

**Issue by dylanrtt** [canjs/canjs#2318]
Given a template like this:

```html
{{#each objects}}
  {{.}}
{{/each}}
```

If you overwrite a key that is already being displayed (e.g., `foo`), the view does not update.

```js
vm.attr('objects.foo', 'blah');
```

http://jsbin.com/kuhexarava/edit?html,js,output

**Comment by justinbmeyer** [comment]
very likely related, might already be working: https://github.com/canjs/canjs/issues/2238

**Comment by justinbmeyer** [comment]
Still a bug: http://jsbin.com/buhosip/edit?html,js,output

**Comment by garrettreed** [comment]
[solved in 3.0](https://github.com/canjs/can-stache/pull/67). 

Implementing the same fix resolves the problem, but breaks other tests. I'll continue to work on it.

**Comment by justinbmeyer** [comment]
Closing for the fix in 3.0.  If someone wants to merge the pull request into the 2.3-legacy branch, we can get it for a future release.
